### PR TITLE
Rest - use jobject to judge whether a json is swagger, no need to deserialize whole file

### PR DESCRIPTION
**Pull request by hellosnow**
use jobject to judge whether a json is swagger, no need to deserialize whole file

@superyyrrzz @chenkennt @vwxyzh @qinezh @ansyral

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.RestApi/RestApiDocumentProcessor.cs [hunk 1]
```diff
@@ -141,11 +142,11 @@ private static bool IsSwaggerFile(string filePath)
         {
             try
             {
-                var dictionary = JsonUtility.Deserialize<Dictionary<string, object>>(filePath);
-                object swaggerValue;
-                if (dictionary.TryGetValue("swagger", out swaggerValue))
+                var jObject = JObject.Parse(File.ReadAllText(filePath));
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.RestApi/RestApiDocumentProcessor.cs]: >Parse [](start = 38, length = 5)

consider: streaming api JObject.Load(new JsonTextReader(TextReader))

Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.RestApi/RestApiDocumentProcessor.cs]: yes it's better.

---
In reply to: [74371215](https://github.com/dotnet/docfx/pull/599#discussion_r74371215) [](ancestors = 74371215)

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.RestApi/RestApiDocumentProcessor.cs [hunk 2]
```diff
@@ -141,14 +142,17 @@ private static bool IsSwaggerFile(string filePath)
         {
             try
             {
-                var dictionary = JsonUtility.Deserialize<Dictionary<string, object>>(filePath);
-                object swaggerValue;
-                if (dictionary.TryGetValue("swagger", out swaggerValue))
+                using (JsonReader reader = new JsonTextReader(new StringReader(File.ReadAllText(filePath))))
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.RestApi/RestApiDocumentProcessor.cs]: >new StringReader(File.ReadAllText [](start = 62, length = 33)

File.OpenText, and using(...)

**Comment by vwxyzh** [comment]
:shipit:

**Pull request merged**